liteflow-labs/starter-kit
# Improve server side error management and redirection to not found errors

### Project organization

- Closes #111 

### Description

- Return not found error on every request to a resource that is empty
- Remove the useless check on `data` (this can never be null/empty), probably from a previous version with wrong typing
- Did not throw any error for any list of items (aka list of assets, list of currency...) -- the app should handle that
- Did not throw any not found error for users, as even if they are not in the database, they can eventually be on the blockchain. Might need the API to always return (and validate) the address, and in that case, the not found will be good to add
- Add missing error management

### How to test

- Go to one of the pages and update the id in the URL, and you should have a 404
  - Checkout page
  - Token page
  - Collection page
  - Create form on a collection

### Checklist
